src/correctness.cpp

```diff
@@ -0,0 +1,120 @@
+#include <array>
+
+#include <piwcsprwmodel/algorithms.h>
+
+#include "correctness.h"
+
+namespace piwcs::prw {
+
+bool isLocallyCorrect(const Model& model, IdRef id){
+	const Node* node = model.node(id);
+
+	std::array<bool, Node::MAX_SLOTS> isInward;
+	std::array<bool, Node::MAX_SLOTS> isOutward;
+	isInward.fill(false);
+	isOutward.fill(false);
+
+	auto sectionAt = [&](SlotId slot) -> const Section*{
+	    return model.section(node->section(slot));
+	};
+
+	if (node->type()==END) {
+		// Single slot of end must be connected to bidir. section or forbidden section
+		return sectionAt(0)->dir()!=Section::AllowedTravel::UNIDIR;
+	}
+
+	else{
+		// Set outward/inward flags
+		for(SlotId start = 0; start < node->sectionCount(); ++start){
+			for(SlotId end = 0; end < node->sectionCount(); ++end){
+
+			    bool isTraversable =
+			            node->couldTraverse(start, end);
+			    bool isBidirectional =
+			            sectionAt(start)->isBidir();
+			    bool isUnidirectional =
+			            sectionAt(start)->isUnidir();
+			    bool isSectionEnd =
+			            sectionAt(start)->end() == id;
+
+				if(isTraversable &&
+				        (isBidirectional || (isUnidirectional && isSectionEnd))
+				        ){
+					isInward[start] = true;
+					isOutward[end] = true;
+				} // if
+			} // for end
+		} // for start
+
+		// Check slot correctness
+		for(SlotId slot = 0; slot < node->sectionCount(); ++slot){
+			const Section* section = sectionAt(slot);
+
+			// Bidirectional slot
+			if(isInward[slot] && isOutward[slot]){
+				if (!section->isBidir()){
+					return false;
+				}
+				continue;
+			}
+
+			// None slot
+			if(!(isInward[slot] || isOutward[slot])) {
+				if(section->allowsTravel()){
+					return false;
+				}
+				continue;
+			}
+
+			// Unidirectional slot
+			if(!section->isUnidir()){
+				return false;
+			}
+
+			bool isUnidirectional = section->isUnidir();
```

Comment:
Create the `isUnidirectional` flag a little earlier to make use of it in the check just above. Also, the comment seems to imply the opposite of what's actually happening...
```suggestion
			bool isUnidirectional = section->isUnidir();
			
			// Unidirectional slot
			if(!isUnidirectional){
				return false;
			}
```

Reply:
Changed in cb58782a014d64ee9e0e9b9ddf00f98707280c2c.